packages/web-runtime/src/container/bootstrap.ts

```diff
@@ -382,13 +364,26 @@ export const announceTranslations = ({
   return gettext
 }
 
-export const announceAdditionalTranslations = ({
+export const announceTranslations = ({
   gettext,
-  translations
+  coreTranslations,
+  customTranslations,
+  appsStore
 }: {
   gettext: Language
-} & Pick<GetTextOptions, 'translations'>): void => {
-  gettext.translations = merge(gettext.translations, translations)
+  coreTranslations: Translations
+  customTranslations?: Translations
+  appsStore?: AppsStore
+}) => {
+  gettext.translations = merge(coreTranslations, customTranslations || {})
+
+  if (appsStore) {
+    loadAppTranslations({
+      apps: appsStore.apps,
+      gettext,
+      lang: gettext.current
```

Comment:
Is this correctly set on application boot? When is the language setting loaded?

Reply:
We need to set it here initially to make sure the correct language has been loaded in an unauthenticated context. In an authenticated context, we set it a second time after the user has been fetched via the Graph API.